pkgs/by-name/zs/zsh-autosuggestions-abbreviations-strategy/package.nix

```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  ...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "zsh-autosuggestions-abbreviations-strategy";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "olets";
+    repo = pname;
+    rev = "83840e5940c0da14ed1e407026b4f26ede18cbf1"; # TODO, CHANGE REV TO 1.1.1 WHEN TAG IS MADE https://github.com/olets/zsh-autosuggestions-abbreviations-strategy/issues/2
+    hash = "sha256-j2Xx8EWcSRntY7gqK9X1/rn3siZgNdL7ht4CyfAA+yY=";
+  };
+
+  installPhase = ''
+    install -D ${pname}.zsh $out/share/zsh/${pname}/${pname}.plugin.zsh
+    install -D ${pname}.zsh $out/share/zsh/${pname}/${pname}.zsh
+  '';
```

Comment:
Do not use `pname` for this. Keep the `pre`/`post` hooks. And from what I remember, the installation directory should be `$out/share/zsh/site-functions`.

```suggestion
  installPhase = ''
    runHook preInstall

    install -D *.zsh "$out/share/zsh/site-functions/"

    runHook postInstall
  '';
```

Reply:
Had to modify this slightly to get it working, due to `install` intricacies. This worked:
```
    install *.zsh -Dt "$out/share/zsh/site-functions/"
```